### PR TITLE
Added sample to Find-and-replace-line-break-in-Word-document-with-paragraph-mark

### PR DESCRIPTION
**Pull request by elizabethatienoo**
Added sample to find and replace line break with paragraph mark

### CODE REVIEW

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        static void Main(string[] args)
+        {
+            using (FileStream fileStreamPath = new FileStream(Path.GetFullPath(@"../../../Data/Input.docx"), FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+            {
+                //Open an existing Word document.
+                using (WordDocument document = new WordDocument(fileStreamPath, FormatType.Docx))
+                {
+                    // Iterate through the paragraphs of Word document.
+                    foreach (WParagraph paragraph in document.Sections[0].Body.Paragraphs)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:17]: Handle the code snippet for the document with multiple sections

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    {
+                        for (int i = 0; i < paragraph.Items.Count; i++)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:19]: Can you check iterating elements in UG and modify the sample as generic requirement.

Sample should replace line break in all areas of document para, table, textbox, content control everything

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        {
+                            Entity entity = paragraph.Items[i];
+                            //Set paragraph child element as break.                                                     
+                            if (entity.EntityType == EntityType.Break)
+                            {
+                                Break breakItem = entity as Break;
+                                //Replace line break with paragraph mark.
+                                if (breakItem.BreakType == BreakType.LineBreak)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:27]: Separate this replace logic alone as separate method

ReplaceLineBreakWithPara()

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                                   //Create new paragraph by cloning the existing paragraph.
+                                    WParagraph newPara = ownerPara.Clone() as WParagraph;
+                                   //Remove the child items after the line break from the old paragraph including line break.
+                                    for (int j = breakIndex; j < ownerPara.ChildEntities.Count;)
+                                    {
+                                        ownerPara.ChildEntities.RemoveAt(j);
+                                    }
+                                    int newParaItemsCount = ownerPara.ChildEntities.Count;
+                                    //Remove the child items before the line break from the new paragraph including line break.
+                                    while (newParaItemsCount + 1 != 0)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:42]: Whether it removing all items in paragraph or items before the line break alone
Could you please check the logic and modify.
Check with test cases document

**Review by MohanaselvamJothi — APPROVED** [review]

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs (hunk 4 above)
Comment (elizabethatienoo) [reply Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs]: It removes the items upto the end of line break so the line break is also being removed

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs (hunk 3 above)
Comment (elizabethatienoo) [reply Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs]: Addressed

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs (hunk 2 above)
Comment (elizabethatienoo) [reply Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs]: Modified

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs (hunk 1 above)
Comment (elizabethatienoo) [reply Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs]: Done

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                        if (breakItem.BreakType == BreakType.LineBreak)
+                        {
+                            WParagraph ownerPara = breakItem.OwnerParagraph;
+                            int breakIndex = ownerPara.ChildEntities.IndexOf(breakItem);
+                            int paraIndex = ownerPara.OwnerTextBody.ChildEntities.IndexOf(ownerPara);
+
+                            //Create new paragraph by cloning the existing paragraph.
+                            WParagraph newPara = ownerPara.Clone() as WParagraph;
+                            //Remove the child items after the line break from the old paragraph including line break.
+                            for (int j = breakIndex; j < ownerPara.ChildEntities.Count;)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:113]: If we not incrementing the index, then why we need to use for loop here?
Can you check and use the valid loop.

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                {
+                    IterateTextBody(cell);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Iterate paragraph child elements.
+        /// </summary>
+        private static void IterateParagraph(ParagraphItemCollection paraItems)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:93]: Can you change method name as "IterateParagraphItems" to convey proper work of method

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                            //Create new paragraph by cloning the existing paragraph.
+                            WParagraph newPara = ownerPara.Clone() as WParagraph;                           
+                            //Remove paragraph child entities from the index of break item.
+                            while (breakIndex <ownerPara.ChildEntities.Count)
+                            {
+                                ownerPara.ChildEntities.RemoveAt(breakIndex);                                
+                            }            
+                            int newParaItemsCount = ownerPara.ChildEntities.Count;
+                            // Remove paragraph child entities from index zero upto the end of paragraph break item.
+                            while (newParaItemsCount + 1 != 0)
```
Comment (MohanaselvamJothi) [inline Find-and-Replace/Replace-line-break-with-paragraph-mark/.NET-Standard/Replace-line-break-with-paragraph-mark/Program.cs:119]: As you aware about the index of break in cloned paragraph, modify logic to remove from 0th index to index of break in cloned paragraph.
Modify while loop to remove items until specific index.

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**